app/models/redirects/ConditionalRedirect.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.redirects
+
+import play.api.mvc.Call
+
+case class ConditionalRedirect(condition: Boolean, redirect: Call, hasPrior: Option[Boolean] = None)
+
+object ConditionalRedirect {
```

Comment:
Unless I've missed something this isn't used anywhere yet. Would it be better to include in a future task when needed?